test/server_handles_broken_connection_test.dart

```diff
@@ -84,4 +100,25 @@ main() async {
       expect(e, isA<grpc.GrpcError>());
     }, reason: 'the client should send an error from the destroyed channel'));
   });
+
+  if (!Platform.isWindows) {
+    test(
```

Comment:
I think the best way to write tests is just to parameterise existing tests, e.g. you have 

```dart
test("the client interrupting the connection does not crash the server",
      () async {
  // ...
});
```

Which hardcodes `localhost` as a host. What you can do is the following

```dart
testTcpAndUds("the client interrupting the connection does not crash the server",
      (InternetAddress bindAddress) async {
  // ... replace occurances of localhost with bindAddress (might require plumbing
  // bindAddress further, e.g. in this particular test into ClientData.
});
```

where the helper function looks like this:

```dart
/// Test functionality for both TCP and Unix domain sockets.
void testTcpAndUds(String name, FutureOr<void> Function(InternetAddress) testCase) {
  test(name, () async {
    final localhostAddresses = await InternetAddress.lookup('localhost');
    await testCase(localhostAddresses.first);
  });
  
  if (Platform.isWindows) {
    return;
  }
  
  test('$name (over uds)', () async {
    final tempDir = await Directory.systemTemp.createTemp();
    final socketAddr = InternetAddress(tempDir.path + '/socket', type: InternetAddressType.unix);
    try {
      await testCase(socketAddr);     
    } finally {
      await tempDir.delete(recursive: true);
    }
  });
}
```

You can then use this helper to enable various tests easily and without much copying. I suggest among other enabling `round trip insecure connection` test.